### PR TITLE
CPV raw writing and reconstruction

### PR DESCRIPTION
**Pull request by peressounko**
Code to write and read CPV raw data, convert raw->digits->clusters->CTF

### CODE REVIEW

**Comment by davidrohr** [comment]
@ktf @TimoWilken @pzhristov @costing : During O2 compilation, we are getting warnings from ROOT about JAlien:
```
Error in <TSystem::ExpandFileName>: input: $ROOT_DYN_PATH/libRIO, output: $ROOT_DYN_PATH/libRIO
Error in <TSystem::ExpandFileName>: input: $ROOT_DYN_PATH/libCling, output: $ROOT_DYN_PATH/libCling
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for TJAlienCollection.h
   requested to autoload type TAliceCollection
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for TJAlienCollection.h
   requested to autoload type TJAlienCollection
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for TJAlienCredentials.h
   requested to autoload type TJAlienCredentials
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for TJAlien.h
   requested to autoload type TJAlienJDL
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for TJAlien.h
   requested to autoload type TAliceFile
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for TJAlien.h
   requested to autoload type TJAlienFile
```

Could it be that some JAlien dictionary is missing? This does not break the build, since we only fail on compiler warnings and this comes from ROOT. But it fills the logs, and I think will be confusing for people when a PR fails and they look for the real compile errors. Any ideas?

**Comment by shahor02** [comment]
@peressounko Please fix what CI complains for:
````
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5129/0/Detectors/CPV/reconstruction/include/CPVReconstruction/RawDecoder.h:31:3: error: use '= default' to define a trivial copy constructor [modernize-use-equals-default]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5129/0/Detectors/CPV/reconstruction/src/RawDecoder.cxx:148:29: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5129/0/Detectors/PHOS/workflow/src/RawToCellConverterSpec.cxx:138:23: error: statement should be inside braces [readability-braces-around-statements]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5129/0/Detectors/CPV/reconstruction/include/CPVReconstruction/RawDecoder.h:31:3: error: use '= default' to define a trivial copy constructor [modernize-use-equals-default]
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/O2/5129/0/Detectors/CPV/reconstruction/include/CPVReconstruction/RawDecoder.h:31:3: error: use '= default' to define a trivial copy constructor [modernize-use-equals-default]
````

File: DataFormats/Detectors/CPV/include/DataFormatsCPV/Cluster.h [hunk 1]
```diff
@@ -66,16 +74,47 @@ class Cluster
...
+    setPackedPosZ(posZ);
+    setPackedEnergy(en);
+    setPackedClusterStatus(status);
+  }
+
  protected:
   char mMulDigit = 0;    ///< Digit nultiplicity
   char mModule = 0;      ///< Module number
   char mNExMax = -1;     ///< number of (Ex-)maxima before unfolding
   int mLabel = -1;       ///< Ref to entry in MCTruthContainer with list of labels
```
Comment (shahor02) [inline DataFormats/Detectors/CPV/include/DataFormatsCPV/Cluster.h]: I see you are setting this label but don't see where it is used. The same for PHOS clusters. Do you need them? Usually we extract MC labels from the container using the cluster index in its container.

**Review by shahor02 — CHANGES_REQUESTED** [review]
@peressounko 
Thanks! Could you also add ``test_ctf_io_cpv.cxx`` similar to ``test_ctf_io_phos.cxx``?

File: DataFormats/Detectors/CPV/include/DataFormatsCPV/Cluster.h (hunk 1 above)
Comment (peressounko) [reply DataFormats/Detectors/CPV/include/DataFormatsCPV/Cluster.h]: Thanks, this is rudiment, removed.

**Comment by shahor02** [comment]
@peressounko the ``CTF/test/test_ctf_io_cpv.cxx`` fails. You can test it locally by e.g.
````
cd alice/sw/BUILD/O2-latest/O2/
./stage/tests/o2-test-ctf-cpv
````

**Comment by peressounko** [comment]
Dear @shahor02, could you please approve this PR to make test of CPV code easier to cpv experts? Thanks, Dmitri

**Comment by ktf** [comment]
Please fix the fullCI issue.

**Comment by shahor02** [comment]
Hi @peressounko , sure, will merge once the tests are passed.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**